### PR TITLE
Merge representation digest definition with Digest header one

### LINKED ISSUE #1833: Consider merging Sections 2 and 3
In @jricher's WGLC feedback, he suggests that section 2 (representation digest) could be merged with section 3 (the digest field). This makes sense since the document has expanded to define also Content-Digest for which representation digests don't apply.

Merging these sections makes editorial sense.

### PR DESCRIPTION
**Pull request by LPardue**
Fixes #1833 

This seems like a better flow to me.

This is a bit hard to review. I'd check the editor's copy built from the branch.

### CODE REVIEW

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -149,9 +148,12 @@ To support use-cases where a simple checksum of the content bytes is required,
 this document introduces the `Content-Digest` request and response header and trailer field; see
 {{content-digest}}.
 
-`Digest` and `Content-Digest` support algorithm agility. The `Want-Digest` and
-`Want-Content-Digest` fields allows endpoints to express interest in `Digest` and
-`Content-Digest` respectively, and preference of algorithms in either.
+`Digest` and `Content-Digest` contain checksums that are calculated using one of
+the digest-algorithms listed in the HTTP Digest Algorithm Values Registry (see
+{{algorithms}}), which are encoded in the format associated with each encoding.
```
Comment (ioggstream) [inline draft-ietf-httpbis-digest-headers.md:153]: encoded...encoding :thinking:

File: draft-ietf-httpbis-digest-headers.md [hunk 2]
```diff
@@ -213,57 +215,42 @@ interpreted as described in {{SEMANTICS}}.
...
-HTTP11}}). To help illustrate how such things affect `Digest`,
-several examples are provided in {{resource-representation}}.
+The `Digest` field contains a comma-separated list of one or more representation
+digest values. It can be used in both requests and responses.
 
-A representation digest consists of
-the value of a checksum computed on the entire selected `representation data`
-(see {{Section 8.1 of SEMANTICS}}) of a resource identified according to {{Section 6.4.2 of SEMANTICS}}
-together with an indication of the algorithm used:
+A representation digest is an integrity mechanism for HTTP resources that uses a
```
Comment (ioggstream) [inline draft-ietf-httpbis-digest-headers.md:223]: I liked having a separate sub/section for that... let me think

File: draft-ietf-httpbis-digest-headers.md (hunk 1 above)
Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:153]:
```suggestion
{{algorithms}}), which are encoded in the format associated with each algorithm.
```

Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:153]: made a typo :)

File: draft-ietf-httpbis-digest-headers.md [hunk 3]
```diff
@@ -213,57 +215,42 @@ interpreted as described in {{SEMANTICS}}.
...
+checksum  calculated independently of the content (see {{Section 6.4 of
+SEMANTICS}}). A representation digest consists of the value of a checksum
+computed on the entire selected `representation data` (see {{Section 8.1 of
+SEMANTICS}}) of a resource identified according to {{Section 6.4.2 of
+SEMANTICS}}, together with an indication of the algorithm used. Note that
+representation data can be fully or partially contained in the content, or not
+contained at all. When a message has no representation data it is possible to
+assert that no representation data was sent by using the representation digest
+on an empty string (see {{usage-in-signatures}}).
```
Comment (ioggstream) [inline draft-ietf-httpbis-digest-headers.md:232]:
```suggestion
checksum  calculated independently of the content (see {{Section 6.4 of
SEMANTICS}}).
A representation digest consists of the value of a checksum
computed on the entire selected `representation data` (see {{Section 8.1 of
SEMANTICS}}) of a resource identified according to {{Section 6.4.2 of
SEMANTICS}}, together with an indication of the algorithm used.
Note that
representation data can be fully or partially contained in the content, or not
contained at all.
When a message has no representation data it is possible to
assert that no representation data was sent by using the representation digest
on an empty string (see {{usage-in-signatures}}).
```

File: draft-ietf-httpbis-digest-headers.md (hunk 2 above)
Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:223]: By merging the section we end up with 9 total paragraphs in one section vs 12 total paragraphs split across 2 sections. That seems editorial cleaner to me.

It also makes the section closer to the layout of Content Digest, which seems beneficial when reading both.

But if you have an alternative proposal I'm happy to review it.

**Review by ioggstream — APPROVED** [review]
LGTM

**Pull request merged**